src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Umbraco.Core.Events;
+using Umbraco.Core.Logging;
+using Umbraco.Web.Common.Lifetime;
+
+namespace Umbraco.Web.Common.Profiler
+{
+    public class InitializeWebProfiling : INotificationHandler<UmbracoApplicationStarting>
+    {
+        private readonly bool _profile;
+        private readonly WebProfiler _profiler;
+        private readonly IUmbracoRequestLifetime _umbracoRequestLifetime;
+        private readonly List<Action> _terminate = new List<Action>();
+        public InitializeWebProfiling(IProfiler profiler, IUmbracoRequestLifetime umbracoRequestLifetime, ILogger<InitializeWebProfiling> logger)
+        {
+            _umbracoRequestLifetime = umbracoRequestLifetime;
+            _profile = true;
+
+            // although registered in WebRuntime.Compose, ensure that we have not
```

Comment:
Is this still relevant?

Reply:
I'll update the comment, but in theory, the profiler can still be replaced by another type before the `UmbracoApplicationStarting` notification is handled